FreeRTOS/FreeRTOS-Plus-TCP
# IPv4/single: Call DNS callback routines while scheduler is not suspended

Description
-----------
This PR is quite similar to my earlier #250., which was merged with [ipv6_multi](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/tree/labs/ipv6_multi).

When using `FreeRTOS_gethostbyname_a()`, an IP address will be looked up "asynchronously". The lookup will be started from the user task, and it will be finished by the IP-task.

When the lookup is ready, or after a time-out, the IP-task will call the DNS application hook, as indicated by the user:
~~~c
uint32_t FreeRTOS_gethostbyname_a(
                const char * pcHostName, // The name of the host to be looked up
                FOnDNSEvent pCallback,   // Application hook, called when ready
                void * pvSearchID,       // An ID that will be passed when calling the hook
                TickType_t uxTimeout );  // A time-out in clock ticks
~~~

While iterating through the list of outstanding DNS lookups, it is necessary to suspend the scheduler: it is possible that `List_t xCallbackList` is accessed from 2 tasks.

Until now, the function `pCallback` was called while the scheduler is suspended. That is not necessary, and it can be disturbing for user.

This PR will change that behaviour: the scheduler will be running while the application hook is called.

Test Steps
-----------
I tested by using `FreeRTOS_gethostbyname_a()` both for a successful lookup, and for a lookup that timed out.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
